action/lib/parse.js

```diff
@@ -24,18 +50,77 @@ export default function (title, target) {
     return process.exit(0) // soft exit
   }
 
+  let isDev = devDependencyRegEx.test(title);
+  let isProd = dependencyRegEx.test(title);
+  const isSecurity = securityRegEx.test(title) || labels.includes("security") || labels.includes("Security");
+
+  if (!isDev && !isProd) {
+    // couldn't extract the dependency type from the title, try to read package.json
+    try {
+      const packageJsonPath = path.join(ghWorkspace, 'package.json')
+      const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, "utf8"));
+      isDev = !!packageJson.devDependencies && depName in packageJson.devDependencies
+      isProd = !!packageJson.dependencies && depName in packageJson.dependencies
+    } catch (e) {
+      console.dir(e);
+     }
+  }
+  if (!isDev && !isProd) {
+    // we failed again, assume its a production dependency
+    core.info(`failed to parse dependency type, assuming it is a production dependency`)
+    isProd = true
+  }
+
   // log
   core.info(`from: ${from.version}`)
   core.info(`to: ${to.version}`)
+  core.info(`dependency type: ${isDev ? "development" : isProd ? "production" : "unknown"}`)
+  core.info(`security critical: ${isSecurity}`)
+
+  // convert target to the automerged_updates syntax
+  const configPath = path.join(ghWorkspace, '.github/auto-merge.yml')
+  let mergeConfig;
+  if (fs.existsSync(configPath)) {
+    // parse .github/auto-merge.yml
+    mergeConfig = yaml.safeLoad(fs.readFileSync(configPath, 'utf8'));
+    core.info('loaded merge config: ' + JSON.stringify(mergeConfig, undefined, 4));
```

Comment:
printing the yaml output as is would be cleaner for debugging purposes (and avoids using `.stringify`)